common/test/test_migrations/utils.py

```diff
@@ -0,0 +1,65 @@
+"""
+General class for Test migrations.For more information visit at
+https://www.caktusgroup.com/blog/2016/02/02/writing-unit-tests-django-migrations/
+"""
+# pylint: disable=redefined-outer-name
+
+from django.test import TransactionTestCase
+from django.db.migrations.executor import MigrationExecutor
+from django.db import connection
+
+
+class TestMigrationsForward(TransactionTestCase):
```

Comment:
I think the right move is to just make one `TestMigrations` class that allows you to forward and backward migrate. `check_data` looks like it can go into `setUp`.

I imagine the methods you'll need are:
- `setUp`
- `migrate_forwards`
- `migrate_backwards`
- `setUpBeforeForwardsMigration`

You can imagine a typical test looking like:

```python
def test_migrate_forward_preserves_thing(self):
    self.migrate_forwards()
    self.assertEqual(self.pre_migration_thing, post_migration_thing)

def test_migrate_backwards_removes_column(self):
    thing = self.create_thing()
    self.migrate_backwards()
    self.assertTrue(column_isnt_there_on_thing(thing))
```
Where `self.migrate_forwards` calls `setUpBeforeForwardsMigration`.

Something else, if we want to have different set ups before forward migrations (say you want to check that something behaves correctly with a `NULL` or non-`NULL` value), we could pass a configuration dict as kwargs to `migrate_forwards`, which would pass it on to `setUpBeforeForwardsMigration`. That way we could use ddt more easily.

So you could imagine:
```python
@ddt.data((True, True), (False, False))
def test_migrate_forward_preserves_thing(self, config1, config2):
    self.migrate_forwards(config1=config1, config2=config2)
    self.assertEqual(self.pre_migration_thing, post_migration_thing)
```